tests/validation/tests/v3_api/test_hpa.py

```diff
@@ -0,0 +1,533 @@
+        ns = rbac_get_namespace()
+        cluster_owner_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+        cluster_owner_client = \
+            get_project_client_for_token(project, cluster_owner_token)
+
+    # Create HPA as a cluster owner
+    hpa, workload = create_hpa(cluster_owner_client, ns)
+    # Verify editing HPA fails
+    with pytest.raises(ApiError) as e:
+        user_client.update(hpa,
+                           name=hpa['name'],
+                           namespaceId=ns.id,
+                           maxReplicas=10,
+                           minReplicas=3,
+                           workload=workload.id,
+                           metrics=metrics)
+    assert e.value.error.status == 403
+    assert e.value.error.code == 'Forbidden'
+
+    remove_resource(hpa)
+    remove_resource(workload)
+
+
+def create_hpa(p_client, ns, workload=None):
+
+    # Create workload of scale 1 with CPU reservation
+    # Create hpa pointing to the workload.
+    if workload is None:
+        workload = create_workload(p_client, ns)
+    name = random_test_name("hpa")
+
+    metrics = [{'name': 'cpu',
+                'type': 'Resource',
+                'target': {
+                    'type': 'Utilization',
+                    'utilization': '50',
+                },
+                }]
+
+    hpa = p_client.create_horizontalPodAutoscaler(
+        name=name,
+        namespaceId=ns.id,
+        maxReplicas=5,
+        minReplicas=2,
+        workloadId=workload.id,
+        metrics=metrics
+    )
+    hpa = wait_for_hpa_to_active(p_client, hpa)
+    assert hpa.type == "horizontalPodAutoscaler"
+    assert hpa.name == name
+    # After hpa becomes active, the workload scale should be equal to the
+    # minReplicas set in HPA object
+    workloadlist = p_client.list_workload(uuid=workload.uuid).data
+    assert workloadlist[0].scale == hpa.minReplicas
+
+    return (hpa, workload)
+
+
+def edit_hpa(p_client, ns):
+
+    # Create workload of scale 1 with memory reservation
+    # Create hpa pointing to the workload.Edit HPA and verify HPA is functional
+
+    workload = create_workload(p_client, ns)
+    name = random_test_name("default")
+
+    metrics = [{
+        "type": "Resource",
+        "name": "memory",
+        "target": {
+            "type": "AverageValue",
+            "value": None,
+            "averageValue": "32Mi",
+            "utilization": None,
+            "stringValue": "32"
+        }
+    }]
+
+    hpa = p_client.create_horizontalPodAutoscaler(
+        name=name,
+        namespaceId=ns.id,
+        maxReplicas=4,
+        minReplicas=2,
+        workloadId=workload.id,
+        metrics=metrics
+    )
+    wait_for_hpa_to_active(p_client, hpa)
+
+    # After hpa becomes active, the workload scale should be equal to the
+    # minReplicas set in HPA
+    workloadlist = p_client.list_workload(uuid=workload.uuid).data
+    assert workloadlist[0].scale == hpa.minReplicas
```

Comment:
Validate that the workload actually has correct number of pods by calling validate_workload

Reply:
Done